src/main/scala/com/knoldus/aws/services/dynamodb/QuestionServiceImpl.scala

```diff
@@ -0,0 +1,65 @@
+package com.knoldus.aws.services.dynamodb
+
+import com.knoldus.aws.models.dynamodb.{ Question, QuestionTable, QuestionUpdate }
+import com.knoldus.aws.utils.Constants
+import com.typesafe.scalalogging.LazyLogging
+import play.api.libs.json.Json
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class QuestionServiceImpl(questionTable: QuestionTable) extends QuestionService with LazyLogging {
+
+  override def submitQuestion(question: Question): Future[Option[String]] =
+    Future {
+      questionTable.put(question.record) match {
+        case Right(id) =>
+          logger.info("Question submitted successfully")
+          Some(Constants.submitQuestionResponse(id))
+        case Left(message) =>
+          logger.error(s"Failed to submitted the question: $message")
+          None
+      }
+    }
+
+  override def getQuestion(id: String, category: String): Future[Option[Question]] =
+    Future {
+      logger.info(s"Getting the question with id: $id and category: $category")
+      questionTable.retrieve(category, id) match {
+        case Some(record) =>
+          logger.info("Question found")
+          Question(Json.parse(record.json)) match {
+            case Left(message) =>
+              logger.error(message)
+              None
+            case Right(question) => Some(question)
+          }
+        case None =>
+          logger.info("Question not found")
+          throw new NoSuchElementException(Constants.noQuestionFoundResponse)
+      }
+    }
+
+  override def updateQuestion(id: String, category: String, questionUpdate: QuestionUpdate): Future[Option[String]] =
+    Future {
+      def updateQuestionEntity(): Question = {
+        val updatedTitle = questionUpdate.title
+        val updatedDescription = questionUpdate.description
+        Question(id, updatedTitle, category, updatedDescription)
+      }
+      logger.info(s"Updating the question with id: $id and category: $category")
+      questionTable.update(category, id, updateQuestionEntity().record) match {
+        case Right(_) =>
+          Some(Constants.updateQuestionResponse)
+        case Left(message) =>
+          logger.error(s"Failed to update the question, reason: $message")
+          throw new NoSuchElementException(Constants.noQuestionFoundResponse)
```

Comment:
If we are directly throwing the exception, then what is the use case of None, we are not returning it anywhere.
If we return the String only, instead of Some(_), then also it works.

Reply:
Will fix that.